types/modules/producer.d.ts

```diff
@@ -0,0 +1,79 @@
+import { Connection } from './connection';
+import { ProducerHooks } from './hooks';
+import type amqp = require('amqplib');
+import pDefer = require('p-defer');
+
+declare class Producer {
```

Comment:
Please private modifiers where appropriate.. `checkRpcQueue` / `maybeAnswer` / etc .. should be private.